include/ocpp/v201/functional_blocks/data_transfer.hpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#pragma once
+
+#include <ocpp/v201/message_dispatcher.hpp>
+#include <ocpp/v201/messages/DataTransfer.hpp>
+
+namespace ocpp {
+namespace v201 {
+
+class DataTransferInterface {
+
+public:
+    /// \brief Sends a DataTransfer.req message to the CSMS using the given parameters
+    /// \param vendorId
+    /// \param messageId
+    /// \param data
+    /// \return DataTransferResponse containing the result from CSMS
+    virtual std::optional<DataTransferResponse> data_transfer_req(const CiString<255>& vendorId,
+                                                                  const std::optional<CiString<50>>& messageId,
+                                                                  const std::optional<json>& data) = 0;
+
+    /// \brief Sends a DataTransfer.req message to the CSMS using the given \p request
+    /// \param request message shall be sent to the CSMS
+    /// \return DataTransferResponse containing the result from CSMS. In case no response is received from the CSMS
+    /// because the message timed out or the charging station is offline, std::nullopt is returned
+    virtual std::optional<DataTransferResponse> data_transfer_req(const DataTransferRequest& request) = 0;
+
+    /// \brief Handles the given DataTransfer.req \p call by the CSMS by responding with a CallResult
+    virtual void handle_data_transfer_req(Call<DataTransferRequest> call) = 0;
```

Comment:
For this functional block not really necessary but for the others it would be nicer if this would just be one call I think. Something like `handle_message()` that internally splits again based on the message type just like the charge_point `handle_message`.

Reply:
Should probably then pass along the `const EnhancedMessage<v201::MessageType>& message`

Reply:
Ah right, I should have added a comment for this since you marked that already in the other PR. I will add it here and definitely agree for other functional blocks, but in the end that is handled/defined by every functional block individually, since the `handle_message` function is not defined in any general interface but in the interface of every functional block.

Reply:
Ah ok no worries. Yes it does not matter that much but thought it would be nice for uniformity ;)

Reply:
I wanted to say the same, it would be nice to have a handle_message in every functional block.

Reply:
I changed this to a `handle_message` function and found it reasonable to add an interface for it. Using that interface would be more of a convention than something that we can enforce when implementing other functional blocks but it provides at least some soft guidance.

This change goes beyond the scope of the PR, but since we are in the the message dispatching / handling is tightly related to the first functional block implementing it, i would be in favor of merging this as part of this PR.